src/main/java/org/openrewrite/hibernate/MigrateUserType.java

```diff
@@ -0,0 +1,258 @@
+                            if (stmt instanceof J.Return) {
+                                J.Return r = (J.Return) stmt;
+                                if (r.getExpression() instanceof J.TypeCast) {
+                                    J.TypeCast tc = (J.TypeCast) r.getExpression();
+                                    if (parameterizedType != null && TypeUtils.isOfType(parameterizedType.getTarget().getType(), tc.getClazz().getType())) {
+                                        return r.withExpression(tc.getExpression());
+                                    }
+                                }
+                            }
+                            return stmt;
+                        })));
+                    }
+                } else if (ASSEMBLE.matches(md, cd) && parameterizedType != null) {
+                    md = md.withReturnTypeExpression(parameterizedType.getTarget().withPrefix(Space.SINGLE_SPACE));
+                    if (md.getReturnTypeExpression() != null) {
+                        md = md.withPrefix(md.getReturnTypeExpression().getPrefix());
+                    }
+                    if (md.getBody() != null) {
+                        md = md.withBody(md.getBody().withStatements(ListUtils.map(md.getBody().getStatements(), stmt -> {
+                            if (stmt instanceof J.Return) {
+                                J.Return r = (J.Return) stmt;
+                                if (r.getExpression() != null && !TypeUtils.isOfType(parameterizedType.getTarget().getType(), r.getExpression().getType())) {
+                                    return r.withExpression(new J.TypeCast(randomId(), Space.EMPTY, Markers.EMPTY, new J.ControlParentheses<>(randomId(), Space.EMPTY, Markers.EMPTY,
+                                            new JRightPadded<>(TypeTree.build("BigDecimal").withType(parameterizedType.getTarget().getType()), Space.EMPTY, Markers.EMPTY)), r.getExpression()));
+                                }
+                            }
+                            return stmt;
+                        })));
+                    }
+                } else if (REPLACE.matches(md, cd) && parameterizedType != null) {
+                    md = md.withReturnTypeExpression(parameterizedType.getTarget().withPrefix(Space.SINGLE_SPACE));
+                    if (md.getReturnTypeExpression() != null) {
+                        md = md.withPrefix(md.getReturnTypeExpression().getPrefix());
+                    }
+                    md = changeParameterTypes(md, Arrays.asList("original", "target"));
+                }
+                updateCursor(md);
+                md = (J.MethodDeclaration) super.visitMethodDeclaration(md, ctx);
+                return maybeAutoFormat(method, md, ctx);
+            }
+
+            private J.MethodDeclaration changeParameterTypes(J.MethodDeclaration md, List<String> paramNames) {
```

Comment:
Matching on parameter names isn't safe here, as folks might have named their variables differently; I suggest you change this to use the parameter index instead, or another way of ensuring only the correct types are changed.

Reply:
Applied parameter index instead of name